src/alda/Util.java

```diff
@@ -43,6 +45,32 @@
     return concat(a, new Object[]{b});
   }
 
+  // Given an array of choices like {"yes", "no", "quit"}, offers them to the
+  // user as choices in this format: (y)es, (n)o, (q)uit
+  //
+  // Returns the selected string as soon as the user presses the corresponding
+  // key.
+  public static String promptWithChoices(List<String> choices)
+  throws IOException {
+    Map<Character, String> m = new LinkedHashMap<Character, String>();
+    for (String choice : choices) {
+      m.put(choice.charAt(0), choice);
+    }
+
+    String choiceString = choices.stream()
+                                 .map(str -> "(" + str.charAt(0) + ")"
+                                             + str.substring(1))
+                                 .collect(Collectors.joining(", "));
+
+    System.out.println(choiceString);
+
+    while (true) {
+      char c = (char) System.in.read();
```

Comment:
I think that reading manually like this might be a bad idea, maybe we could pass in a `ConsoleRead` object into this function, so that all the fancy readline features from jline still work in these prompts (ie: ctrl-u, ctrl-w, ctrl-a, ctrl-e, etc). (or maybe expose one in a static public class somewhere)?